docs/docs/guides/wallet/signing.md

```diff
@@ -0,0 +1,130 @@
+---
+sidebar_position: 4
+sidebar_label: 'Tutorial: Signing operations'
+---
+
+# Signing operations
+
+In this guide, we'll cover how to sign `data` and `transactions` using web3.js. Whether it's signing data with an account, signing transactions, or using wallets and private keys, we'll walk you through the basic steps. Let's get started!
+
+
+## Sign Data With an `Account`
+
+In this example we are creating a random account `web3.eth.account.create()`, but you can also import a specific account by using `web3.eth.accounts.privateKeyToAccount('0x...')`
+
+``` ts
+import { create } from 'web3-eth-accounts';
+
+// the create method returns a `Web3Account` object
+// the account contains an `address` and `privateKey` and allows you to be able to encrypt, signData and signTransaction.
+const account = create(); //this is the same as web3.eth.accounts.create();
+
+/* ↳
+{
+address: '0xbD504f977021b5E5DdccD8741A368b147B3B38bB',
+privateKey: 'privateKey',
+signTransaction: [Function: signTransaction],
+sign: [Function: sign],
+encrypt: [AsyncFunction: encrypt]
+}
+*/
+
+account.sign('hello world');
+/* ↳ 
+{
+  message: 'hello world',
+  messageHash: '0xd9eba16ed0ecae432b71fe008c98cc872bb4cc214d3220a36f365326cf807d68',
+  v: '0x1b',
+  r: '0xe4fce466ef18f6cd8b4f4175a9a04cd2872a1a6a8cfc2ff67fb0cfd6d78ec758',
+  s: '0x37ca3a789976f1854d16e50a04caf2e06ee14b0ac4a5878b43929767f2008288',
+  signature: '0xe4fce466ef18f6cd8b4f4175a9a04cd2872a1a6a8cfc2ff67fb0cfd6d78ec75837ca3a789976f1854d16e50a04caf2e06ee14b0ac4a5878b43929767f20082881b'
+}
+*/
+```
+
+## Sign a Transaction With an `Account`
+
+In this example we are importing a specific account by using `web3.eth.accounts.privateKeyToAccount('0x...')` but you can also create a random account by using `web3.eth.account.create()`.
+
+``` ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3(/* PROVIDER */);
+
+const privateKey = '0x4651f9c219fc6401fe0b3f82129467c717012287ccb61950d2a8ede0687857ba'
+const account = web3.eth.accounts.privateKeyToAccount(privateKey);
+
```

Comment:
What about adding a not similar to the on for sendTransaction?
```suggestion

// Magic happens inside signTransaction. If a transaction is sent from an account that exists in a wallet, it will be automatically signed.

```